mods/classic-clock-button-behavior.wh.cpp

```diff
@@ -0,0 +1,90 @@
+// ==WindhawkMod==
+// @id              classic-clock-button-behavior
+// @name            Classic Clock Button Behavior
+// @description     Makes the tray clock open the Date and Time CPL on double click
+// @version         1.0.0
+// @author          aubymori
+// @github          https://github.com/aubymori
+// @include         explorer.exe
+// @architecture    x86-64
+// ==/WindhawkMod==
+
+// ==WindhawkModReadme==
+/*
+# Classic Clock Button Behavior
+In Windows XP and before, the tray clock did nothing on single click, and opened
+the Date and Time CPL on double click. This mod makes the Windows 10 tray clock
+button do that.
+*/
+// ==/WindhawkModReadme==
+
+#include <windhawk_utils.h>
+#include <windows.h>
+
+constexpr UINT_PTR DOUBLE_CLICK_TIMER_ID = 5500;
+
+bool g_bDoubleClicked = false;
+
+void CALLBACK TimerProc(HWND hWnd, UINT uMsg, UINT_PTR nIDEvent, DWORD dwTime)
+{
+    g_bDoubleClicked = false;
+    KillTimer(hWnd, nIDEvent);
+}
+
+#define ClockButton_Window(pThis) *((HWND *)pThis + 1)
+
+HRESULT (*ClockButton_v_OnClick_orig)(void *, UINT);
+HRESULT ClockButton_v_OnClick_hook(
+    void *pThis,
+    UINT  uClickDevice
+)
+{
+    if (g_bDoubleClicked)
+    {
+        g_bDoubleClicked = false;
+        ShellExecuteW(
+            ClockButton_Window(pThis),
+            NULL,
+            L"timedate.cpl",
+            NULL,
+            NULL,
+            SW_SHOWNORMAL
+        );
+    }
+    else
+    {
+        g_bDoubleClicked = true;
+        SetTimer(
```

Comment:
You could just save the first click time and avoid the timer. Here's an example:
https://github.com/ramensoftware/windhawk-mods/blob/bf5b0746e2cb9d22e109b664ca3a63c430199f18/mods/pinned-items-double-click.wh.cpp#L113-L131

Just a suggestion, let me know if you want to adapt it or have it merged as is.

Reply:
It's fine as is.